pkg/expr/func.go

```diff
@@ -82,27 +87,44 @@ func newNEQ() Func {
 	}
 }
 
-func (f *eqFunc) Call(args []interface{}) interface{} {
-	res := f.call(args)
-	if f.invert {
-		return !res
+func (f *eqFunc) Call(attrs attribute.Bag, args []*Expression, fMap map[string]FuncBase) (interface{}, error) {
+	arg0, err := args[0].Eval(attrs, fMap)
+	if err != nil {
+		return nil, err
 	}
-	return res
-}
 
-func (f *eqFunc) call(args []interface{}) bool {
-	var s0 string
-	var s1 string
-	var ok bool
+	arg1, err := args[1].Eval(attrs, fMap)
+	if err != nil {
+		return nil, err
+	}
+
+	res := f.call(arg0, arg1)
 
-	if s0, ok = args[0].(string); !ok {
-		return reflect.DeepEqual(args[0], args[1])
+	if f.invert {
+		return !res, nil
 	}
-	if s1, ok = args[1].(string); !ok {
-		// s0 is string and s1 is not
-		return false
+	return res, nil
+}
+
+func (f *eqFunc) call(args0 interface{}, args1 interface{}) bool {
+	switch s0 := args0.(type) {
+	default:
+		return reflect.DeepEqual(args0, args1)
+	case bool:
```

Comment:
do we ever imagine different behaviors for bool, int64, and float64?  if not, should we collapse the cases to be:
```
switch ... {
case bool, int64, float64:
   return args0 == args1
...
}
```